src/xtgeo/common/sys.py

```diff
@@ -22,6 +23,34 @@
 from ._xyz_enum import _AttrType
 from .xtgeo_dialog import XTGeoDialog
 
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    import pandas as pd
+
+    from xtgeo import (
+        BlockedWell,
+        BlockedWells,
+        Cube,
+        Grid,
+        GridProperties,
+        GridProperty,
+        Polygons,
+        RegularSurface,
+        Well,
+    )
+
+    XTGeoObject = Union[
+        BlockedWell,
+        BlockedWells,
+        Cube,
+        Grid,
+        GridProperty,
+        GridProperties,
+        Polygons,
+        RegularSurface,
+        Well,
+    ]
```

Comment:
This might be something good to move to `xtgeo.types` when it's merged in. I will do it when also adapting `GridProperty` to use FileLike from it later (this list is also incomplete)